City-of-Helsinki/drupal-hdbt
# Set the aspect-ratio of cookie compliance block to 16/9

* Set aspect ratio for cookie complicance blocks so that it takes the same amount of space than the video element would to reduce jumping and page change when cookies are accepted.
* Also set max-width of the paragraphs to content-max-width to avoid too long text lines.

![image](https://user-images.githubusercontent.com/1191667/141119420-71b72fbc-7ca7-4939-a5d1-fe165c2eadae.png)

## How to test

* Get branch
* Add a video embed to a page
* Open page with cookies not accepted
* Notice the grey background elements from cookie message mimicing the video element dimensions

## Known limitations
The `aspect-ratio` is not supported on iOS and MacOS Safari before version 15, but the fallback is graceful, the box size is defined by the content size and if it's supported, longer content is allowed to override the aspect ratio. This makes this solution better than hacks like the `padding-top: 56.5%;`
